packages/api/src/providers/fitbit.ts

```diff
@@ -373,7 +388,86 @@ export class Fitbit extends Provider implements OAuth2 {
     );
   }
 
-  // Creates a subscription for all or one specific collectionType
+  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
+  async revokeTokenFromExistingUsers(
+    userId: string,
+    fitbitUserId: string,
+    activeSubscriptions: string[],
+    accessToken: string
+  ) {
+    try {
+      if (activeSubscriptions.length > 0) {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+        await Promise.all(
+          Object.values(connectedUsers).map(async user => {
+            if (user.dataValues.id !== userId) {
+              await this.oauth.revokeLocal(user);
+              user = await getConnectedUserOrFail({
+                id: user.id,
+                cxId: user.cxId,
+              });
+              sendProviderDisconnected(user, [ProviderSource.fitbit]);
+            }
+          })
+        );
+        await this.deleteSubscriptions(activeSubscriptions, fitbitUserId, accessToken);
```

Comment:
trying to wrap my head around this... why do we delete WH subscriptions for the current user that just connected to afterwards create subscriptions again?

Reply:
So this handles the scenario where we have 2 metriport users connecting to the same Fitbit user. 
User A previously connected. User B connects now. We want to remove all of the WH subscriptions for the Fitbit user, and re-create them for user B because of the potential difference in the scopes that they allowed. Otherwise, if user A subbed for all of `["sleep", "activities", "nutrition", "body"], but user B only wants to sub to `"sleep"`, we would not keep the Fitbit user subbed to all of these types. This is "a solution", but, perhaps, not the most elegant one. Maybe it's better to identify differences in scopes and remove existing unnecessary subs, and add new non-existing subs. Thoughts?